backend/server.js

```diff
@@ -23,6 +60,44 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+// Create user - sign up
+app.post('/users', async (req, res) => {
+  try {
+    const { name, email, password } = req.body
+    const user = new User({ name, email, password: bcrypt.hashSync(password) })
+    const saved = await user.save()
+    res.status(201).json(saved)
+  } catch (err) {
+    res.status(401).json({ message: 'Could not create user', errors: err.errors })
+  }
+})
+
+// Go get one users info/profile page
+app.get('/users/:id', authenticateUser)
+app.get('/users/:id', async (req, res) => {
+  try {
+    res.status(201).json(req.user)
+  } catch (err) {
+    res.status(400).json({ message: 'could not save user', errors: err.message })
+  }
+})
+
+// Log in
+app.post('/sessions', async (req, res) => {
+  try {
+    const { email, password } = req.body
+    const user = await User.findOne({ email })
+    if (user && bcrypt.compareSync(password, user.password)) {
+      res.json({ userId: user._id, accessToken: user.accessToken })
+    } else {
+      res.json({ notFound: true, message: 'wrong username or password' })
```

Comment:
Maybe a status code here as well? :)

Reply:
And since you're using the email as login in the frontend (and not the username), write that here as well.